pcl_ros/CMakeLists.txt

```diff
@@ -118,8 +119,19 @@ ament_target_dependencies(pcl_ros_tf
 #
 ### Tools
 #
-#add_executable(pcd_to_pointcloud tools/pcd_to_pointcloud.cpp)
-#target_link_libraries(pcd_to_pointcloud ${Boost_LIBRARIES} ${catkin_LIBRARIES} ${PCL_LIBRARIES})
+add_library(pcd_to_pointcloud_lib SHARED tools/pcd_to_pointcloud.cpp)
+target_link_libraries(pcd_to_pointcloud_lib
+  ${PCL_LIBRARIES})
+target_include_directories(pcd_to_pointcloud_lib PUBLIC
```

Comment:
I meant that more as a suggestion to actually add a `main` file option so that users don't have to only interact with this as a component (e.g. publish me a pointcloud for a test). I try to have any components also have an executable counterpart so that users can use either that they wish that is most sensible for their workflow.

Reply:
Oh ok, I actually did not know that. Is that documented somewhere / new?

Reply:
It's documented [as part of the docstring](https://github.com/ros2/rclcpp/blob/03fa731d2345c41e5a4e436bf0730012ece96ce5/rclcpp_components/cmake/rclcpp_components_register_node.cmake#L15-L16). The feature was added in 2019, so not terribly new. I'm not sure if we call it out in any tutorials, but a lot of our demo packages use it.

Reply:
Ah feels like something that should be explicitly mentioned in composition tutorials. Here nor there for this PR, but I'd suggest maybe spending a few minutes if you can think of a place that this note would be good to expose to users. Many people's complaints about ROS 2 are these "hidden" features or got-ya's that aren't documented. Small actions accumulated over time make a big difference :-)

Reply:
Unfortunately, I don't think we have a tutorial on writing and building node components. I'm only aware of [this tutorial](https://docs.ros.org/en/rolling/Tutorials/Composition.html) for interacting with components that already exist on the system.

Reply:
... there's no "how do you make a component" tutorial?

Reply:
I suppose https://docs.ros.org/en/rolling/Concepts/About-Composition.html?highlight=rclcpp_components_register_node#writing-a-component and https://docs.ros.org/en/rolling/Tutorials/Actions/Writing-a-Cpp-Action-Server-Client.html?highlight=rclcpp_components_register_node#writing-an-action-server-and-client-c are as good of a place as currently exists.

Reply:
Yeah, as @jacobperron mentions, this already is creating an executable.
The [PR description](https://github.com/ros-perception/perception_pcl/pull/359#issue-1190196659) has some instructions of how to use it.

> I suppose https://docs.ros.org/en/rolling/Concepts/About-Composition.html?highlight=rclcpp_components_register_node#writing-a-component and https://docs.ros.org/en/rolling/Tutorials/Actions/Writing-a-Cpp-Action-Server-Client.html?highlight=rclcpp_components_register_node#writing-an-action-server-and-client-c are as good of a place as currently exists.

Yeah, I agree this should be explicitly mentioned in some of the tutorials.
Currently we're using it in many of the demos though, https://github.com/ros2/demos/blob/f008dbbeefd7da30c4133ad67655554d9f7f8519/demo_nodes_cpp/CMakeLists.txt#L112-L114.